### DISCUSSION TITLE
Help on selecting audio device

### DISCUSSION THREAD

**Discussion by Shellfishgene** [tphakala/birdnet-go#21] [General]
Hi!

Birdnet-go seems very nice, thanks for making it!
I have not found any info in the documentation on how exactly to select the audio device to use for recording, on Linux or Windows. On Windows it seems to use my webcam, but I also have a separate mic. On the Raspberry I have a USB soundcard, but there is also the builtin one. I seems to pick the right one, but I'm not sure how to actually check.

Thanks!

**Comment by tphakala** [comment]
Hello and thanks for feedback!

At the moment there is no option to select audio device on application itself, BirdNET-Go uses default capture device of your OS. I will add option to choose capture device on later releases.

To change default capture device on Windows open "change system sounds"

![Snag_95390c](https://github.com/tphakala/birdnet-go/assets/7030001/e228aaf8-6844-43b2-994d-652e68a84af3)

and set audio device you wish to use as default

![Snag_954179](https://github.com/tphakala/birdnet-go/assets/7030001/bc3d1974-59b3-48f2-849a-ea59f062b188)

On Linux Pulse Audio audio source is recommended, if you have pulse audio installed it should default automatically to capture device on USB port as Raspberry Pi does not have built-in audio capture device. If you are using latest Bookworm release of Raspberry Pi OS install pipewire-pulse package for capture to work

```
sudo apt install pipewire-pulse
sudo reboot
```

If these steps do not work please run BirdNET-Go with -d flag to enable debug and provide its output

```
birdnet-go[.exe] realtime -d
```

**Comment by Shellfishgene** [comment]
Thank you very much for the help! I hadn't realized the Raspberry can't record through its 3.5 mm jack.
Do you also plan to write a frontend for birdnet-go, like BirdNET-Pi? That one does not seem to be developed further anymore.

**Reply by tphakala** [reply]
Yes, web ui is much needed and I am infact working on it right now. Here is how it looks currently. I need to polish it bit more before I can include it in public release, but this my priority focus at the moment.

![Snag_161f1fc](https://github.com/tphakala/birdnet-go/assets/7030001/aafe5e49-1f37-4bd4-8e66-4f8a7f2fdc6a)

**Reply by Shellfishgene** [reply]
Wow, looks great already, looking forward to using it! As a side note, I am very happy I don't have to learn bird names in Finnish ;)

**Reply by tphakala** [reply]
I have now pushed code containing web UI to github, if you are willing to compile code yourself you can start testing it. To enable spectrograms you need to install sox cli tool, "apt install sox" or so depending on your os.

**Reply by Shellfishgene** [reply]
Thanks! It compiled, but I get this error:
`Command execution error: failed to initialize BirdNET: failed to load labels: label file 'labels_.txt' not found in the zip archive`
I assume it's not reading the config file correctly? I have the locale set to "de".

**Reply by tphakala** [reply]
Sorry, locale setting should be under birdnet: node now, you need to update config file manually or maybe better to let app create new config with updated options by renaming old config file.

**Reply by Shellfishgene** [reply]
My mistake, didn't use the new config file. It works now, will have to wait for some birds tomorrow to check it out (don't want to move the microphone for testing now).

**Reply by Shellfishgene** [reply]
Cool, the setting are in the UI, too. :)

**Reply by Shellfishgene** [reply]
I noticed one bug (?), it did not create the spectrograms directory automatically, and then went in a loop creating and not finding the spectrograms of the recent detections. Once I created the directory it worked.
But it's very nice! And even themes are planned. Do you know yet how you will implement the statistics?

**Reply by tphakala** [reply]
Yep there might be lot of small issues like spectrogram directory not being created automatically, I will fix them though. There is already light and dark themes, is it switched automatically based on browser (and/or OS) preference.

Regarding statistics, I have lots of ideas. One of things I will implement soon is to add open weather map integration for fetching weather data with sunrise and sunset times. With this it will be possible to filter nocturnal species out during daytime as vice versa, it also allows some behavioral statistics based on weather conditions.
